tests/test_path_checks.py

```diff
@@ -95,3 +103,66 @@ def test_unicode_in_output_files(tmp_path: Path, filename: str) -> None:
         filename,
     ]
     assert main(params) == 0
+
+
+def test_clt_returns_specialchar_names(tmp_path: Path) -> None:
```

Comment:
This needs a docstring: https://github.com/common-workflow-language/cwltool/pull/1446/checks?check_run_id=2668401511#step:7:26